common/scala/src/main/scala/org/apache/openwhisk/http/PoolingRestClient.scala

```diff
@@ -110,9 +110,16 @@ class PoolingRestClient(
       if (response.status.isSuccess) {
         Unmarshal(response.entity.withoutSizeLimit).to[T].map(Right.apply)
       } else {
-        // This is important, as it drains the entity stream.
-        // Otherwise the connection stays open and the pool dries up.
-        response.discardEntityBytes().future.map(_ => Left(response.status))
+        Unmarshal(response.entity).to[String].flatMap { body =>
+          val statusCode = response.status
+          val reason =
+            if (body.length > 0) s"${statusCode.reason} (details: $body)" else statusCode.reason
```

Comment:
Please use `body.nonEmpty`. Computing the whole length of a string (or any datastructure really) **can** be O(n), whereas `isEmpty`/`nonEmpty` is **always** O(1)